applications/CompressiblePotentialFlowApplication/custom_elements/embedded_compressible_potential_flow_element.cpp

```diff
@@ -123,6 +123,11 @@ ModifiedShapeFunctions::Pointer EmbeddedCompressiblePotentialFlowElement<2,3>::p
     return Kratos::make_shared<Triangle2D3ModifiedShapeFunctions>(this->pGetGeometry(), rDistances);
 }
 
+template <>
+ModifiedShapeFunctions::Pointer EmbeddedCompressiblePotentialFlowElement<3,4>::pGetModifiedShapeFunctions(Vector& rDistances) {
+    return Kratos::make_shared<Tetrahedra3D4ModifiedShapeFunctions>(this->pGetGeometry(), rDistances);
```

Comment:
Couldn't this be a unique pointer?

Reply:
Do you say this because we are not copying the pointer anywhere?  I guess you are right! I'm not sure if I completely understand the use-cases of shared vs unique pointers.